sql/core/src/test/scala/org/apache/spark/sql/streaming/FlatMapGroupsWithStateSuite.scala

```diff
@@ -795,7 +795,7 @@ class FlatMapGroupsWithStateSuite extends StateStoreMetricsTest {
         }
       },
       CheckNewAnswer(("c", "-1")),
-      assertNumStateRows(total = 0, updated = 0)
+      assertNumStateRows(total = 0, updated = 1)
```

Comment:
This represents the patch is correct fix, though total = 0, updated = 1 might make someone feeling odd.

Reply:
are there other tests where we can check this fix as well?

Reply:
Actually the change we encountered the issue is small, as the bug is affecting the metric only when no data batch updates the state row. This is not occurred in non-arbitrary stateful operations, as possible state update is only eviction here, and state row eviction is not counted as updates.

This can be occurred in arbitrary stateful operations as timer can be triggered for no data batch and the query can update/delete state row which would trigger update count. This test is verifying one of these cases (delete state), hence I didn't add the new test. If we would like to have another test for state func to update the state row on timeout I can do, but most of things would be redundant.